### PR TITLE
Reorganise code structure

### PR DESCRIPTION
**Pull request by zalewskise**
### What has changed?
 - Moved code into separate files and packages 
 - Renamed class `Store` to `ReduxStore` and interface `StoreType` to `Store`
 
 ### Why it has changed?
  - To (hopefully) increase code readability :)

### CODE REVIEW

File: cored/src/commonMain/kotlin/com/github/kittinunf/cored/store/DefaultStore.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    return Store(
+        scope = scope,
+        initialState = initialState,
+        engine = DefaultEngine(
+            reducer = combineReducers(reducer, SetStateReducer()),
+            middlewares = middlewares.toMutableList()
+        )
+    )
+}
+
```
Comment (kittinunf-dd) [inline cored/src/commonMain/kotlin/com/github/kittinunf/cored/store/DefaultStore.kt:57]: 👀

File: cored/src/commonTest/kotlin/com/github/kittinunf/cored/ReduxStoreCreationTest.kt [hunk 2]
```diff
@@ -14,7 +14,7 @@ class ReduxStoreCreationTest {
 
     private val testScope = CoroutineScope(Dispatchers.Unconfined)
 
-    private val store = Store(testScope, CounterState(),
+    private val store = com.github.kittinunf.cored.store.Store(testScope, CounterState(),
```
Comment (kittinunf-dd) [inline cored/src/commonTest/kotlin/com/github/kittinunf/cored/ReduxStoreCreationTest.kt:17]: 👀

**Comment by zalewskise** [comment]
@kittinunf 
I would also like to suggest following name changes:
-  `EffectType` -> `Effect`
- `ReducerType` -> `ActionReducer`

Naming is hard and I think `Type` doesn't really gives any value in it :) 

Also: 
- `DefaultEngine` -> `IteratorEngine`
- `MapBackedEngine` -> `HashEngine`

File: cored/src/commonMain/kotlin/com/github/kittinunf/cored/Reducer.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package com.github.kittinunf.cored
+
+import kotlin.reflect.KClass
+
+typealias AnyReducer<S> = Reducer<S, Any>
+typealias ReducerType<S, A> = Pair<KClass<out Any>, Reducer<S, A>>
+
+enum class Order {
+    BeforeReduce,
+    AfterReduced
```
Comment (zalewskise) [inline cored/src/commonMain/kotlin/com/github/kittinunf/cored/Reducer.kt]: `AfterReduced` or `AfterReduce`? 🤔

Comment (kittinunf) [reply cored/src/commonMain/kotlin/com/github/kittinunf/cored/Reducer.kt]: Yeah ... it think it should have been `AfterReduce`*

**Comment by kittinunf** [comment]
Ready for review?

**Comment by zalewskise** [comment]
Changed the:
`EffectType` -> `ActionMiddleware` to be consistent with Reducer naming :)

**Comment by kittinunf** [comment]
Ohhhh mate I like this naming! ❤️ 👍 🦞 🚢

**Pull request merged**